### ISSUE TITLE
FATAL EXCEPTION: Failed resolution of: Lcom/mapbox/android/telemetry/NavigationMetadata;

### ISSUE THREAD

**Issue by sjd753** [mapbox/mapbox-navigation-android#1987]
**Android API:** 

  minSdkVersion 17
  compileSdkVersion 28
  targetSdkVersion 28

**Mapbox Navigation SDK version:**

implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:8.1.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.40.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.40.0'

### Steps to trigger behavior

 1. Run the mapbox navigation ui in android 28 or above
 2. Play with the ui for a little bit 
 3. App crashes with the below logs

![image](https://user-images.githubusercontent.com/14920076/60580369-9a453380-9da2-11e9-9a83-d58f19039aa0.png)

### Expected behavior
App / Mapbox navigation runs smoothly
### Actual behavior
App crashes all of a sudden while in navigation mode

**Comment by iamdeveloper-lopez** [comment]
@sjd753 If you are using 

> implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.40.0' 

you should remove the 

> implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:8.1.0'
> implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.40.0'

Because NavigationUI sdk implemented already the mapbox-android-sdk & mapbox-android-navigation

Try to remove the two sdk then try to run again. 

Hope this help you.

**Comment by Guardiola31337** [comment]
Hey @sjd753 👋 thanks for reaching out and report your issue.

@iamdeveloper-lopez is totally right.

If you're using the Navigation UI SDK, you don't have to declare the Mapbox Maps SDK, it's included implicitly 👀 https://github.com/mapbox/mapbox-navigation-android/blob/156e6a7aafe3bb603ac7c0ba28a322e0925b7a8d/libandroid-navigation-ui/build.gradle#L49 This is on purpose to avoid these kind of dependencies conflicts.

That being said, we’re always striving to upgrade to the latest versions of the different dependencies included in the SDK. This issue specifically is caused because the Events version included in Mapbox Maps SDK doesn't include the Navigation events 👀 https://github.com/mapbox/mapbox-events-android/pull/383 - this was included in Events [`v4.5.0`](https://github.com/mapbox/mapbox-events-android/releases/tag/telem-4.5.0-core-1.3.0) and Events `v4.5.1` landed in Maps `v8.1.0` https://github.com/mapbox/mapbox-gl-native/pull/14874 (any previous versions work 👌). We're tracking the work needed in the Navigation SDK side in https://github.com/mapbox/mapbox-navigation-android/pull/1890

There's another workaround that could be made in the meantime which is excluding the Events dependency from Maps SDK so that Events `v4.4.1` from the Navigation SDK is used 👀 

```gradle
implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:8.1.0') {
  exclude group: 'com.mapbox.mapboxsdk', module: 'mapbox-android-telemetry'
}
```

Closing as answered. Please feel free to cut a new ticket if you run into any other issues. Thanks again!
